### PR TITLE
[MP] reconcile game module changes with the original source releases by diffing codemp/game

### PR DESCRIPTION
**Pull request by Razish**
Main changes:

* revert saberMoveData changes to match basejka, provide a new cvar (`g_fixSaberMoveData`) to patch it if desired
* introduce a new configstring (26: `CS_LEGACY_FIXES`) to inform the client of which fixes it should also apply, e.g. saberMoveData
* ~client module obtains gamestate (including configstrings) earlier to prevent use before initialisation, which is now checked for in debug builds~ reverted due to bugs
* revert a change to saber disarm bonus calculation if `g_fixSaberDisarmBonus` is not enabled
* `G_CanBeEnemy` considers private duels once again (removed from jasdk_modbase before its git history)

### CODE REVIEW

File: codemp/game/bg_saber.c [hunk 1]
```diff
@@ -316,30 +322,23 @@ saberMoveData_t	saberMoveData[LS_MOVE_MAX] = {//							NB:randomized
...
 	// Parry
 	{"Parry Top",	BOTH_P1_S1_T_,		Q_R,	Q_T,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_BL2TR,		LS_A_T2B,		150	},	// LS_PARRY_UP,
 	{"Parry UR",	BOTH_P1_S1_TR,		Q_R,	Q_TL,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_BL2TR,		LS_A_TR2BL,		150	},	// LS_PARRY_UR,
 	{"Parry UL",	BOTH_P1_S1_TL,		Q_R,	Q_TR,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_BR2TL,		LS_A_TL2BR,		150	},	// LS_PARRY_UL,
-	{"Parry LR",	BOTH_P1_S1_BR,		Q_R,	Q_BR,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_TL2BR,		LS_A_BR2TL,		150	},	// LS_PARRY_LR,
-	{"Parry LL",	BOTH_P1_S1_BL,		Q_R,	Q_BL,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_TR2BL,		LS_A_BL2TR,		150	},	// LS_PARRY_LL
-	//{"Parry LR",	BOTH_P1_S1_BL,		Q_R,	Q_BR,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_TL2BR,		LS_A_BR2TL,		150	},	// LS_PARRY_LR,
-	//{"Parry LL",	BOTH_P1_S1_BR,		Q_R,	Q_BL,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_TR2BL,		LS_A_BL2TR,		150	},	// LS_PARRY_LL
+	{"Parry LR",	BOTH_P1_S1_BL,		Q_R,	Q_BR,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_TL2BR,		LS_A_BR2TL,		150	},	// LS_PARRY_LR,
+	{"Parry LL",	BOTH_P1_S1_BR,		Q_R,	Q_BL,	AFLAG_ACTIVE,	50,		BLK_WIDE,	LS_R_TR2BL,		LS_A_BL2TR,		150	},	// LS_PARRY_LL
```
Comment (Razish) [inline codemp/game/bg_saber.c:341]: This has been reverted back to the SDK values. The patched values came from OJP.
We now apply the patch at runtime based on `g_fixSaberMoveData`

File: codemp/game/g_bot.c [hunk 2]
```diff
@@ -139,6 +139,9 @@ int G_GetMapTypeBits(char *type)
 			typeBits |= (1 << GT_TEAM);
 			typeBits |= (1 << GT_JEDIMASTER);
 		}
+		if( strstr( type, "team" ) ) {
+			typeBits |= (1 << GT_TEAM);
+		}
```
Comment (Razish) [inline codemp/game/g_bot.c:144]: This is actually from Asteroids.

File: codemp/game/g_cmds.c [hunk 3]
```diff
@@ -2358,6 +2358,12 @@ void Cmd_CallTeamVote_f( gentity_t *ent ) {
 	char	arg1[MAX_CVAR_VALUE_STRING] = {0};
 	char	arg2[MAX_CVAR_VALUE_STRING] = {0};
 
+	if ( g_gametype.integer < GT_TEAM )
+	{
+		trap->SendServerCommand( ent-g_entities, "print \"Cannot call a team vote in a non-team gametype!\n\"" );
+		return;
+	}
+
```
Comment (Razish) [inline codemp/game/g_cmds.c:2366]: This is actually from Asteroids.

File: codemp/game/w_saber.c [hunk 4]
```diff
@@ -109,6 +109,16 @@ qboolean G_CanBeEnemy( gentity_t *self, gentity_t *enemy )
...
 
+	if (self->client->ps.duelInProgress && self->client->ps.duelIndex != enemy->s.number)
+	{ //dueling but not with this person
+		return qfalse;
+	}
+
+	if (enemy->client->ps.duelInProgress && enemy->client->ps.duelIndex != self->s.number)
+	{ //other guy dueling but not with me
+		return qfalse;
+	}
```
Comment (Razish) [inline codemp/game/w_saber.c:120]: This was removed from jasdk_modbase before its git history. We don't know why, so let's hope bringing it back is the right move.

File: codemp/game/bg_saber.c [hunk 5]
```diff
@@ -349,6 +348,38 @@ saberMoveData_t	saberMoveData[LS_MOVE_MAX] = {//							NB:randomized
...
+#elif defined(_CGAME)
+	const char *cs = CG_ConfigString(CS_LEGACY_FIXES);
+	const uint32_t legacyFixes = strtoul(cs, NULL, 0);
+	const qboolean doFix = !!(legacyFixes & (1 << LEGACYFIX_SABERMOVEDATA));
+#endif
+	saberMoveData_t *move;
+
+	for (move = saberMoveData; move - saberMoveData < ARRAY_LEN(saberMoveData); move++) {
+		if (!strcmp(move->name, "BParry LR")) {
+			move->animToUse = doFix ? BOTH_H1_S1_BR : BOTH_H1_S1_BL;
```
Comment (SomaZ) [inline codemp/game/bg_saber.c:363]: Maybe just return early on !doFix instead?

Comment (Razish) [reply codemp/game/bg_saber.c:363]: Nope, since the server can currently toggle it back and forth. We want to be able to restore the original value.

**Comment by SomaZ** [comment]
So, what's missing, or what do you plan to add? How's the current state of this?

**Comment by Razish** [comment]
This is ready to go.
Oops, didn't realise I still had the big scary label still applied.

I have two follow-up branches locally that build on top of this

**Comment by taysta** [comment]
Have been running this in TaystJK for quite some time, we had issues until we got the gamestate earlier as already applied to this PR. Have not had any issues since.

**Review by SomaZ — APPROVED** [review]

**Pull request merged**